src/plugins/auto_batch/src/plugin.cpp

```diff
@@ -19,344 +15,382 @@
-    auto&& d = deviceWithBatch;
-    auto openingBracket = d.find_first_of('(');
-    auto closingBracket = d.find_first_of(')', openingBracket);
-    auto deviceName = d.substr(0, openingBracket);
+DeviceInformation Plugin::parse_batch_device(const std::string& device_with_batch) {
+    auto openingBracket = device_with_batch.find_first_of('(');
+    auto closingBracket = device_with_batch.find_first_of(')', openingBracket);
+    auto deviceName = device_with_batch.substr(0, openingBracket);
 
     int batch = 0;
     if (closingBracket != std::string::npos && openingBracket < closingBracket) {
-        batch = std::stol(d.substr(openingBracket + 1, closingBracket - 1));
+        batch = std::stol(device_with_batch.substr(openingBracket + 1, closingBracket - 1));
 
         if (batch <= 0) {
-            IE_THROW() << "Batch value for '" << deviceName << "' must be > 0, while " << batch << "is passed";
+            OPENVINO_THROW("Batch value for '", deviceName, "' must be > 0, while ", batch, "is passed");
         }
     }
-    return {deviceName, {{}}, batch};
+    return {deviceName, {{}}, static_cast<uint32_t>(batch)};
 }
 
-DeviceInformation Plugin::ParseMetaDevice(const std::string& devicesBatchCfg,
-                                          const std::map<std::string, std::string>& user_config) const {
-    auto metaDevice = ParseBatchDevice(devicesBatchCfg);
-    metaDevice.config = GetCore()->GetSupportedConfig(metaDevice.device_name, user_config);
-
+DeviceInformation Plugin::parse_meta_device(const std::string& devices_batch_config,
+                                            const ov::AnyMap& user_config) const {
+    auto meta_device = parse_batch_device(devices_batch_config);
+    meta_device.device_config = get_core()->get_supported_property(meta_device.device_name, user_config);
     // check that no irrelevant config-keys left
     for (const auto& k : user_config) {
         const auto& name = k.first;
-        if (metaDevice.config.find(name) == metaDevice.config.end() &&
+        if (meta_device.device_config.find(name) == meta_device.device_config.end() &&
             !ov::util::contains(supported_configKeys, name)) {
-            IE_THROW() << "Unsupported config key: " << name;
+            OPENVINO_THROW("Unsupported config key: ", name);
         }
     }
-    return metaDevice;
+    return meta_device;
 }
 
-InferenceEngine::RemoteContext::Ptr Plugin::CreateContext(const InferenceEngine::ParamMap& remote_properties) {
-    auto cfg = remote_properties;
-    auto it = cfg.find(CONFIG_KEY(AUTO_BATCH_DEVICE_CONFIG));
-    if (it == cfg.end())
-        it = cfg.find(ov::device::priorities.name());
-    if (it == cfg.end())
-        IE_THROW() << "Value for KEY_AUTO_BATCH_DEVICE_CONFIG is not set";
+ov::SoPtr<ov::IRemoteContext> Plugin::create_context(const ov::AnyMap& remote_properties) const {
+    auto full_properties = remote_properties;
```

Comment:
I think auto batch should know the name of device under the plugin.
Remove properties can be passed without device name.

Reply:
@ilyachur How the plugin know the device name? From compile model when plugin call compile_model()?

Reply:
@ilyachur ilya, about the remote context in auto batch:
1) I don't see explicit scenario of user call like this:
    core->create_context("BATCH", {...}) or, core->get_default_context("BATCH", {...}) also this is previously defined behavior here:
https://github.com/openvinotoolkit/openvino/blob/master/src/inference/src/core.cpp#L278
https://github.com/openvinotoolkit/openvino/blob/master/src/inference/src/core.cpp#L291
2) do you see any other usage like in proxy plugin we need to explictly make api calls above?
3) considering above, currently scenario I think it's enough that batch only support remote context in compiled model. if we really want to extend this support to BATCH, it should be something like this:
core->create_context("BATCH:GPU", {other_property_for_gpu}) 
and core->get_default_context("BATCH:GPU")
and then parse the hardware devices in BATCH and return context for the hardware devices

Reply:
> 

name of device under batch plugin can vary and is been set on-the-fly, e.g.
std::async(compile_model("BATCH:GPU"));
std::async(compile_model("BATCH:VPU"));

Reply:
> I think auto batch should know the name of device under the plugin. Remove properties can be passed without device name.

also if need to implement the create context api in batch,  considering another threading case:
thread 1: core.set_property("BATCH", ov::device::priority("GPU"));
               *****
               core.create_context("BATCH")
thread 2: core.set_property("BATCH", ov::device::priority("VPU"))  // suppose that vpu support remote context
                ******
               core.create_context("BATCH")
in this case, if we do not parse device as a parameter in properties, we cannot guarentee that BATCH plugin can create expected correct context, e,g: gpu context for thread 1 and vpu context for thread 2

Reply:
one more thing, with the implementation of get_default_context(), caused some gpu func test failed, which caused by no find the device throw exception.
![image](https://github.com/openvinotoolkit/openvino/assets/52686861/14d828ec-8713-4633-b8b8-1d7b758de446)